substrate/frame/metadata-hash-extension/src/utils.rs

```diff
@@ -0,0 +1,138 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Function to convert hex string to Option<[u8; 32]> in a const context.
+/// Returns `None` if fails to decode.
+pub const fn hex_str_to_32_bytes_panic(hex_str: &str) -> Option<[u8; 32]> {
```

Comment:
Why is there `panic` in the name? Same for the other helper fn.

Reply:
former implementation, fixed in https://github.com/paritytech/polkadot-sdk/pull/5277/commits/edffdafaae14648761d203f11d9bf52d68a61363